terminology/terminology_test.go

```diff
@@ -0,0 +1,597 @@
+		"version": "1.0.0",
+		"compose": {
+			"include": [
+				{
+					"system": "http://example.org/fhir/CodeSystem/test"
+				}
+			]
+		}
+	}`
+
+	vsSystemWithFilter := `{
+		"resourceType": "ValueSet",
+		"url": "http://example.org/fhir/ValueSet/system-with-filter",
+		"version": "1.0.0",
+		"compose": {
+			"include": [
+				{
+					"system": "http://example.org/fhir/CodeSystem/test",
+					"filter": [
+						{
+							"property": "concept",
+							"op": "is-a",
+							"value": "parent-concept"
+						}
+					]
+				}
+			]
+		}
+	}`
+
+	return []string{vsSystemFilter, vsSystemWithFilter}
+}
+
+func TestExpandValueSet(t *testing.T) {
+	tests := []struct {
+		name                string
+		valueSetURL         string
+		version             string
+		expectedCodes       []*Code
+		expectError         bool
+		expectSpecificError error
+	}{
+		{
+			name:        "existing expansion",
+			valueSetURL: "http://example.org/fhir/ValueSet/with-expansion",
+			version:     "",
+			expectedCodes: []*Code{
+				{System: "http://example.org/fhir/CodeSystem/expansion", Code: "expansion-code1", Display: "Expansion Code 1"},
+				{System: "http://example.org/fhir/CodeSystem/expansion", Code: "expansion-code2", Display: "Expansion Code 2"},
+			},
+		},
+		{
+			name:        "compose with direct concepts",
+			valueSetURL: "http://example.org/fhir/ValueSet/child1",
+			version:     "",
+			expectedCodes: []*Code{
+				{System: "http://example.org/fhir/CodeSystem/child1", Code: "child1-code1", Display: "Child 1 Code 1"},
+				{System: "http://example.org/fhir/CodeSystem/child1", Code: "child1-code2", Display: "Child 1 Code 2"},
+			},
+		},
+		{
+			name:        "nested ValueSet references",
+			valueSetURL: "http://example.org/fhir/ValueSet/parent",
+			version:     "",
+			expectedCodes: []*Code{
+				{System: "http://example.org/fhir/CodeSystem/parent", Code: "parent-code1", Display: "Parent Code 1"},
+				{System: "http://example.org/fhir/CodeSystem/child1", Code: "child1-code1", Display: "Child 1 Code 1"},
+				{System: "http://example.org/fhir/CodeSystem/child1", Code: "child1-code2", Display: "Child 1 Code 2"},
+				{System: "http://example.org/fhir/CodeSystem/child2", Code: "child2-code1", Display: "Child 2 Code 1"},
+				{System: "http://example.org/fhir/CodeSystem/grandchild", Code: "grandchild-code1", Display: "Grandchild Code 1"},
+			},
+		},
+		{
+			name:                "circular references",
+			valueSetURL:         "http://example.org/fhir/ValueSet/circular",
+			version:             "",
+			expectError:         true,
+			expectSpecificError: ErrCircularReference,
+		},
+	}
+
+	// Create provider with the test ValueSets
+	provider, err := NewInMemoryFHIRProvider(getMainTestValueSets())
+	if err != nil {
+		t.Fatalf("Failed to create provider: %v", err)
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			codes, err := provider.ExpandValueSet(tt.valueSetURL, tt.version)
+			
+			if tt.expectError {
```

Comment:
For all of these tests could you move the error case to it's own test func? I've found that test logic is a lot more readable and easy to maintaine when we do a different test grouping for error cases.